portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -19,14 +19,39 @@
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import java.util.ArrayList;
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
-
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
+    // response.setContentType("text/html;");
+    // response.getWriter().println("<h1>Hello Khadija!</h1>");
+    ArrayList<String> comments = new ArrayList<String>();
+    comments.add("Welcome to my page!");
+    comments.add("Here are own personal comments");
+    comments.add(" Have a great time browsing and finsing out more about me.");
+    String json = convertToJson(comments);
+    response.setContentType("application/json:");
+    response.getWriter().println(json);
   }
+public String convertToJson(ArrayList<String> comment) {
```

Comment:
If this works for now, great, but a hard-coded solution like this will be hard to extend later if you want to add more comments. I see you already added the gson dependency in your pom.xml file, so you can import com.google.gson.Gson to this file and use its convertToJsonUsingGson() function instead of writing your own. See the step 3 json walkthrough for an example.